From a340b6039d4a32acaa2c438d0fa7e4d76caeb237 Mon Sep 17 00:00:00 2001
From: Henry Fang <46612090+HenryF23@users.noreply.github.com>
Date: Tue, 3 Dec 2024 16:33:08 -0800
Subject: [PATCH 12/20] Update the INSTALL.md file to include steps for
 deploying al-folio on Netlify. (#2798)

Update CUSTOMIZE.md to include steps to deploy al-folio on Netlify.

---------

Co-authored-by: Henry <Henry@home.com>
Co-authored-by: Henry_Lab <henry@lab.com>
---
 INSTALL.md | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/INSTALL.md b/INSTALL.md
index 9bebeae2..d597ceb7 100644
--- a/INSTALL.md
+++ b/INSTALL.md
@@ -129,6 +129,26 @@ If you keep your site on another branch, open `.github/workflows/deploy.yml` **o
 
 If you need to manually re-deploy your website to GitHub pages, go to Actions, click "Deploy" in the left sidebar, then "Run workflow."
 
+### Deploy on [Netlify](https://www.netlify.com/)
+
+1. [Use this template -> Create a new repository](https://github.com/new?template_name=al-folio&template_owner=alshedivat).
+2. Netlify: **Add new site** -> **Import an existing project** -> **GitHub** and give Netlify access to the repository you just created.
+3. Netlify: In the deploy settings
+
+   - Set **Branch to deploy** to `main`
+   - **Base directory** is empty
+   - Set **Build command** to `sed -i "s/^\(baseurl: \).*$/baseurl:/" _config.yml && bundle exec jekyll build`
+   - Set **Publish directory** to `_site`
+
+4. Netlify: Add the following two **environment variables**
+
+   - | Key            | Value                                                                                  |
+     | -------------- | -------------------------------------------------------------------------------------- |
+     | `JEKYLL_ENV`   | `production`                                                                           |
+     | `RUBY_VERSION` | set to the Ruby version found in `.github/workflows/deploy.yml` (for example, `3.3.5`) |
+
+5. Netlify: Click **Deploy** and wait for the site to be published. If you want to use your own domain name, follow the steps in [this documentation](https://docs.netlify.com/domains-https/custom-domains/).
+
 ### Deployment to another hosting server (non GitHub Pages)
 
 If you decide to not use GitHub Pages and host your page elsewhere, simply run:
-- 
2.48.1

